core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -701,10 +716,16 @@ void finishBecauseUnavailable(ShardId shardId, String message) {
         }
     }
 
-    protected Releasable getIndexShardOperationsCounter(ShardId shardId) {
+    protected IndexShardReference getIndexShardReferenceOnPrimary(ShardId shardId) {
```

Comment:
nite: can we add some java docs about the life cycle of this reference (closed when operation is done and replication have completed).